wgpu-hal/src/metal/command.rs

```diff
@@ -592,7 +592,27 @@ impl crate::CommandEncoder<super::Api> for super::CommandEncoder {
         _offset: u32,
         _data: &[u32],
     ) {
-        //TODO
+        if _stages.contains(wgt::ShaderStages::COMPUTE) {
```

Comment:
This needs to be a tiny bit more complex.
The way it's supposed to work is if you have `[u8; max_push_constants]` array on the command encoder. `set_push_constants` would just overwrite some of these bytes, but the shaders would see as much as `PipelineLayout` have declared.

E.g. user could 0..4 with one `set_push_constants` and 4..8 with another call, and the shader needs to see all of them.